src/release/breaking-changes/material-3-migration.md

```diff
@@ -0,0 +1,515 @@
+        letterSpacing: 0.0,
+      ),
+  ),
+),
+```
+
+[`TextTheme`]: {{site.api}}/flutter/material/TextTheme-class.html
+
+### Components
+
+Some components couldn't merely be updated to match the Material 3 Design spec
+but needed a whole new implementation. Such components require manual migration
+since the Flutter SDK doesn’t know what, exactly, you want.
+
+Replace the Material 2 style [`BottomNavigationBar`][] widget with the new
+[`NavigationBar`][] widget. It’s slightly taller, contains pill-shaped
+navigation indicators, and uses new color mappings.
+
+Code before migration:
+
+```dart
+BottomNavigationBar(
+  items: const <BottomNavigationBarItem>[
+    BottomNavigationBarItem(
+      icon: Icon(Icons.home),
+      label: 'Home',
+    ),
+    BottomNavigationBarItem(
+      icon: Icon(Icons.business),
+      label: 'Business',
+    ),
+    BottomNavigationBarItem(
+      icon: Icon(Icons.school),
+      label: 'School',
+    ),
+  ],
+),
+```
+
+Code after migration:
+
+```dart
+NavigationBar(
+  destinations: const <Widget>[
+    NavigationDestination(
+      icon: Icon(Icons.home),
+      label: 'Home',
+    ),
+    NavigationDestination(
+      icon: Icon(Icons.business),
+      label: 'Business',
+    ),
+    NavigationDestination(
+      icon: Icon(Icons.school),
+      label: 'School',
+    ),
+  ],
+),
+```
+
+Check out the complete sample on
+[migrating from BottomNavigationBar from NavigationBar][].
+
+Replace the [`Drawer`][] widget with [`NavigationDrawer`][], which provides
+pill-shaped navigation indicators, rounded corners, and new color mappings.
+
+Code before migration:
+
+```dart
+Drawer(
+  child: ListView(
+    children: <Widget>[
+      DrawerHeader(
+        child: Text(
+          'Drawer Header',
+          style: Theme.of(context).textTheme.titleLarge,
+        ),
+      ),
+      ListTile(
+        leading: const Icon(Icons.message),
+        title: const Text('Messages'),
+        onTap: () { },
+      ),
+      ListTile(
+        leading: const Icon(Icons.account_circle),
+        title: const Text('Profile'),
+        onTap: () {},
+      ),
+      ListTile(
+        leading: const Icon(Icons.settings),
+        title: const Text('Settings'),
+        onTap: () { },
+      ),
+    ],
+  ),
+),
+```
+
+Code after migration:
+
+```dart
+NavigationDrawer(
+  children: <Widget>[
+    DrawerHeader(
+      child: Text(
+        'Drawer Header',
+        style: Theme.of(context).textTheme.titleLarge,
+      ),
+    ),
+    const NavigationDrawerDestination(
+      icon: Icon(Icons.message),
+      label: Text('Messages'),
+    ),
+    const NavigationDrawerDestination(
+      icon: Icon(Icons.account_circle),
+      label: Text('Profile'),
+    ),
+    const NavigationDrawerDestination(
+      icon: Icon(Icons.settings),
+      label: Text('Settings'),
+    ),
+  ],
+),
+```
+
+Check out the complete sample on [migrating from Drawer from NavigationDrawer][].
```

Comment:
There are two "froms" here. Consider something like:

```
For more guidance, check out another sample of
[migrating from `Drawer` to `NavigationDrawer`][].
```

(Do upgrade the below link definition as well to whatever you decide :)